File: pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

int get_height()
{
    int height = -1;

    while (height < 0 || height > 23)
    {
        height = get_int("Height: ");
    }

    return height;
}

void print_n(int n, char c)
{
    for (int i = 0; i < n; i ++)
    {
        printf("%c", c);
    }
}

void print_pyramid(int height)
{
    char BLANK = ' ';
    char BLOCK = '#';

    int blocks = 1;

    while (blocks < height + 1)
    {
        int blanks = height - blocks;

        print_n(blanks, BLANK);
        print_n(blocks, BLOCK);
        print_n(2, BLANK);
        print_n(blocks, BLOCK);
        printf("\n");

        blocks += 1;
    }
}

int main(void)
{
    int height = get_height();
    print_pyramid(height);
}

File: pset6/mario/mario.py
from cs50 import get_int

BLANK = ' '
BLOCK = '#'


def main():
    height = get_int("Height: ")
    while height < 0 or height > 23:
        height = get_int("Height: ")

    blocks = 1
    for i in range(height):
        blanks = height - blocks
        print_row(blanks, blocks)
        blocks += 1


def print_row(outside, inside):
    print_n(outside, BLANK)
    print_n(inside, BLOCK)
    print_n(2, BLANK)
    print_n(inside, BLOCK)
    print()


def print_n(n, c):
    for i in range(n):
        print(c, end='')


if __name__ == "__main__":
    main()